yamscripts.yml

```diff
@@ -10,54 +10,46 @@ scripts:
   # Run the activity-stream mochitests with the browser toolbox debugger.
   # Often handy in combination with adding a "debugger" statement in your
   # mochitest somewhere.
-  mochitest-debug: (cd $npm_package_config_mc_dir && ./mach mochitest --jsdebugger browser/extensions/activity-stream/test/functional/mochitest )
+  mochitest-debug: (cd $npm_package_config_mc_dir && ./mach mochitest --jsdebugger browser/extensions/activity-stream/test/functional/mochitest)
 
-  cleanmc: rimraf $npm_package_config_mc_dir/browser/extensions/activity-stream
+# bundle: Build all assets for activity stream
+  bundle:
+    locales: pontoon-to-json --src locales --dest data
+    webpack: webpack --config webpack.system-addon.config.js --display-optimization-bailout
+    css: node-sass --source-map true --source-map-contents content-src/styles -o css
+    html: rimraf prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js
 
 # buildmc: Export the bootstraped add-on to mozilla central
   buildmc:
-    pre: =>cleanmc
-    locales: pontoon-to-json --src locales --dest system-addon/data
-    webpack: webpack --config webpack.system-addon.config.js --display-optimization-bailout
-    css: node-sass --source-map true --source-map-contents system-addon/content-src/styles -o system-addon/css
-    html: rimraf system-addon/prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js
-    # Copy over all of the system-addon directory
-    copy: cpx "system-addon/**/{,.}*" $npm_package_config_mc_dir/browser/extensions/activity-stream
+    pre: rimraf $npm_package_config_mc_dir/browser/extensions/activity-stream/
+    bundle: => bundle
+    copy: rsync --exclude-from .mcignore -a . $npm_package_config_mc_dir/browser/extensions/activity-stream/
```

Comment:
If we want the m-c version of the repository to be accurate, we do indeed want to copy over dot files including `.eslintrc`; however:
```
$ ./mach lint browser/extensions/activity-stream/
An error occurred running eslint. Please check the following error messages:


Oops! Something went wrong! :(

ESLint: 4.19.1.
ESLint couldn't find the plugin "eslint-plugin-import". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-import is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-import@latest --save-dev
```

I don't think we want to require m-c eslint to require our additional plugins yet. I suppose one simple fix is to have `m-c/.eslintignore` ignore all of `browser/extensions/activity-stream` ?? I believe @Standard8 has mentioned it's more acceptable for us as we run it ourselves. Although for anyone who tries to modify activity-stream directly from m-c won't get linting coverage…

Reply:
Alternatively, we could just .mcignore our .eslintrc.js for now… @Standard8 any suggestions?

Reply:
I tried ignoring `.eslintrc.js`, although for some reason i'm having trouble properly excluding files in `.eslintignore` (either a local one or the global one) when running ./mach lint

Reply:
Ok, for some reason it wasn't working before, applying this patch seems to do the trick:
```diff
diff --git a/.eslintignore b/.eslintignore
index a7846d0e5630..c19e4a2d497a 100644
--- a/.eslintignore
+++ b/.eslintignore
@@ -93,6 +93,10 @@ browser/extensions/pocket/content/panels/js/vendor/**
 browser/extensions/activity-stream/data/content/activity-stream.bundle.js
 browser/extensions/activity-stream/test/**
 browser/extensions/activity-stream/vendor/**
+browser/extensions/activity-stream/*.js
+browser/extensions/activity-stream/bin/**
+browser/extensions/activity-stream/loaders/**
+browser/extensions/activity-stream/content-src/**
 # The only file in browser/locales/ is pre-processed.
 browser/locales/**
 # imported from chromium
```

Reply:
Ok, so looks like we also needed to update python lint configuration

```diff
diff --git a/.flake8 b/.flake8
index 4df557d37236..ff91d4f4e460 100644
--- a/.flake8
+++ b/.flake8
@@ -3,6 +3,7 @@
 ignore = E121, E123, E126, E129, E133, E226, E241, E242, E704, W503, E402, E741
 max-line-length = 99
 exclude =
+    browser/extensions/activity-stream/node_modules/,
     browser/extensions/mortar/ppapi/,
     build/pymake/,
     testing/mochitest/pywebsocket,
```

and cpplint?
```diff
diff --git a/tools/lint/cpp-virtual-final.yml b/tools/lint/cpp-virtual-final.yml
index 319e971ac68b..f8f8ccb12779 100644
--- a/tools/lint/cpp-virtual-final.yml
+++ b/tools/lint/cpp-virtual-final.yml
@@ -10,6 +10,7 @@ cpp-virtual-final:
         - gfx/skia
         - media/webrtc/trunk
         - security/sandbox/chromium
+        - browser/extensions/activity-stream/node_modules
     extensions: ['cpp', 'h', 'mm']
     type: regex
     #
```

Reply:
we still lint all of `lib/` and `common/` I guess :P but if we're ok with ignoring the whole thing, that's probably fine too